### DISCUSSION TITLE
babel not work after change node_modules location

### DISCUSSION THREAD

**Discussion by monic-shixi** [babel/babel#13359] [Q&A] [answered]
Hi, I don't known that is a bug or my issue.

work version location for `node_modules`: `workspace/projectA/node_modules`
after change, location is: `workspace/cache/projectA/node_modules`

how babel to require presets and plugins in `.babelrc.js` ?

i had add prefix for presets and plugins with new location:

original:

```js
plugins: [
  pluginsA: 'plugin-name'
]
```

after change:

```js
const nodePath = path.resolve(__dirname, '../../cache/projectA/node_modules')
plugins: [
  pluginsA: `${nodePath}/plugin-name`
]
```

ps: webpack with resolveLoader.modules: [nodePath]

but this not work, with some error:
`Must export a default export when using ES6 modules.` with thread-loader.

So, i want to known, babel how to load presets and plugins.

**Comment by nicolo-ribaudo** [comment, marked as answer]
Babel loads presets and plugins relatively to the `.babelrc.js` file that specifies them. For example if `.babelrc.js` is in `/user/dev/my-app/workspace/projectA/.babelrc.js`, Babel will check in:
- `/user/dev/my-app/workspace/projectA/node_modules`
- `/user/dev/my-app/workspace/node_modules`
- `/user/dev/my-app/node_modules`
- `/user/dev/node_modules`
- `/user/node_modules`
- `/node_modules`

As you noticed, you can overwrite Babel's resolution algorithm by manually pre-resolving the plugins and presets path (for example with `path.resolve` and string concatenation, as you did).

**Reply by monic-shixi** [reply]
> Babel loads presets and plugins relatively to the `.babelrc.js` file that specifies them. For example if `.babelrc.js` is in `/user/dev/my-app/workspace/projectA/.babelrc.js`, Babel will check in:
> 
> * `/user/dev/my-app/workspace/projectA/node_modules`
> * `/user/dev/my-app/workspace/node_modules`
> * `/user/dev/my-app/node_modules`
> * `/user/dev/node_modules`
> * `/user/node_modules`
> * `/node_modules`
> 
> As you noticed, you can overwrite Babel's resolution algorithm by manually pre-resolving the plugins and presets path (for example with `path.resolve` and string concatenation, as you did).

Thank you very much!
